src/_pytest/nodes.py

```diff
@@ -417,6 +429,42 @@ def __init__(
 
         super().__init__(name, parent, config, session, nodeid=nodeid, fspath=fspath)
 
+        self._norecursepatterns = self.config.getini("norecursedirs")
+
+    def gethookproxy(self, fspath):
+        from _pytest.main import Session  # noqa: F811
+        from _pytest.python import Package
+
+        if not isinstance(self, (Session, Package)):
```

Comment:
Sorry, I know you already discussed this, but I couldn't quite follow, hope you don't mind me asking again.

Why not add it in the base class as `_gethookproxy` and then change `def gethookproxy` in `Session` and `Package` to just call `self._gethookproxy(fspath)`? Then it's private (underscored) for everyone, except the two subclasses who "opt in" to make it public for backward compat.

Reply:
Good idea!

Reply:
Amended.  (Note to self: old HEAD: da6edba24)